### PR TITLE
Add image downloading to Makefile. Assorted directory changes

### PR DESCRIPTION
**Pull request by dansbecker**
@sallamander Can you look over the changes to the Makefile and let me know if you are ok with them?

If so, you'll avoid merge conflicts if you merge this branch into yours before changing the Makefile.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -9,66 +9,73 @@
...
 
 ##############
-# ILSVRC2014 # 
+# ILSVRC2014 #
 ##############
 
-# Training Bounding Boxes # 
-data/ILSVRC/bb/ILSVRC2014_DET_bbox_train.tgz: 
+# Training Bounding Boxes #
+work/.imagenet_training_bbox_sentinel:
```
Comment (dansbecker) [inline Makefile:44]: We're now getting rid of the tgz files as discussed in gitter and using sentinels. Sentinels are created at very last step, so that if something errors (e.g. tar xvf fails), we don't create a misleading sentinel suggesting the operation was completed.

File: Makefile [hunk 2]
```diff
@@ -9,66 +9,73 @@
...
+work/.imagenet_training_images_sentinel: code/data_setup/download_images.py data/fall11_imagenet_urls.txt
+	python code/data_setup/download_images.py "training"
+	touch work/.imagenet_training_images_sentinel
+
+work/.imagenet_validation_images_sentinel: code/data_setup/download_images.py data/fall11_imagenet_urls.txt
+	python code/data_setup/download_images.py "validation"
+	touch work/.imagenet_validation_images_sentinel
+
+
+imagenet_bb: work/ILSVRC/parsed_training_bb.csv work/ILSVRC/parsed_validation_bb.csv
```
Comment (dansbecker) [inline Makefile:87]: this replaces ILSVRC.  I find the name imagenet easier to remember.  If you think we're going to get more imagenet datasets later on and we want to disambiguate, let's think of a way to do that.

File: Makefile [hunk 3]
```diff
@@ -9,66 +9,73 @@
...
 
-# Validation Bounding Boxes # 
-data/ILSVRC/bb/ILSVRC2013_DET_bbox_val.tgz: 
-	curl http://image-net.org/image/ilsvrc2013/ILSVRC2013_DET_bbox_val.tgz \
-		-o data/ILSVRC/bb/ILSVRC2013_DET_bbox_val.tgz
 
-	tar xvf data/ILSVRC/bb/ILSVRC2013_DET_bbox_val.tgz
-
-	mv ILSVRC2013_DET_bbox_val/ data/ILSVRC/bb/validation
-
```
Comment (sallamander) [inline Makefile:60]: Did you mean to delete the validation download here? If so, why? If not, I can merge this in and make sure to update it when I make my changes.

File: Makefile [hunk 4]
```diff
@@ -9,66 +9,73 @@
...
-	
-work/ILSVRC/parsed_training_bb.csv: 
+work/ILSVRC/parsed_training_bb.csv: work/.imagenet_training_bbox_sentinel
 	bash code/data_setup/parse_xml.sh data/ILSVRC/bb/training/ \
 		work/ILSVRC/parsed_training_bb.csv
 
-work/ILSVRC/parsed_validation_bb.csv: 
+work/ILSVRC/parsed_validation_bb.csv: work/.imagenet_validation_bbox_sentinel
 	bash code/data_setup/parse_xml.sh data/ILSVRC/bb/validation/ \
 				work/ILSVRC/parsed_validation_bb.csv
```
Comment (sallamander) [inline Makefile:68]: So the bbox csvs go here. We should probably change the ILSVRC here to imagenet as well. I don't see us getting more data, but if we do I assume we could just differentiate it with another folder (like the year or something).

Comment (dansbecker) [inline Makefile:68]: Cool.  We can do it as a separate PR.

File: Makefile (hunk 3 above)
Comment (dansbecker) [inline Makefile:60]: Good catch. That was accidental (literally forgot to hit save again before making the commit).
